libethereum/BlockChain.cpp

```diff
@@ -1188,37 +1192,49 @@ void BlockChain::garbageCollect(bool _force)
 	m_lastCollection = chrono::system_clock::now();
 
 	Guard l(x_cacheUsage);
-	WriteGuard l1(x_blocks);
-	WriteGuard l2(x_details);
-	WriteGuard l3(x_blockHashes);
-	WriteGuard l4(x_receipts);
-	WriteGuard l5(x_logBlooms);
-	WriteGuard l6(x_transactionAddresses);
-	WriteGuard l7(x_blocksBlooms);
 	for (CacheID const& id: m_cacheUsage.back())
 	{
 		m_inUse.erase(id);
 		// kill i from cache.
 		switch (id.second)
 		{
 		case (unsigned)-1:
-			m_blocks.erase(id.first);
-			break;
+			{
+				WriteGuard l1(x_blocks);
+				m_blocks.erase(id.first);
+				break;
+			}
 		case ExtraDetails:
-			m_details.erase(id.first);
-			break;
+			{
+				WriteGuard l2(x_details);
+				m_details.erase(id.first);
+				break;
+			}
+		// TODO: ExtraBlockHash seems missing.
```

Comment:
Yes, let's add it, with `m_blockHashes` & `x_blockHashes`

Reply:
@pirapira addressed?